code/controllers/subsystem/aura.dm

```diff
@@ -0,0 +1,98 @@
+/// Tracks aura emitters and tells them to pulse. Also used for creating new ones.
+SUBSYSTEM_DEF(aura)
+	name = "Auras"
+	wait = 2 SECONDS
+	flags = SS_NO_INIT
+	///Stores currently active aura_bearer datums
+	var/list/active_auras = list() //We can't use a normal subsystem processing list because as soon as an aura_bearer leaves the list it needs to die
+
+/datum/controller/subsystem/aura/fire(resumed)
+	for(var/datum/aura_bearer/bearer AS in active_auras)
+		bearer.process()
+
+///Use this to start a new emitter with the specified stats. Returns the emitter in question, use stop_emitting() to end early.
+/datum/controller/subsystem/aura/proc/add_emitter(atom/center, type, range, strength, duration)
+	if(!istype(center))
+		return
+	if(!type || !range || !strength)
+		return
+	return new /datum/aura_bearer(center, type, range, strength, duration)
+
+///The thing that actually pushes out auras to nearby mobs.
+/datum/aura_bearer
+	///What we emit from
+	var/atom/center
+	///String indicating aura variety
+	var/aura_type
+	///How far from our center we apply
+	var/range
+	///Multiplier to aura effectiveness
+	var/strength
+	///How many ticks we have left, 15 by default.
+	var/duration
+	///If our aura_type is in this, we'll pulse to humans
+	var/list/human_auras = list(AURA_HUMAN_MOVE, AURA_HUMAN_HOLD, AURA_HUMAN_FOCUS)
```

Comment:
uh, shouldn't this be a const or at the very least a static?

Reply:
can't be a const since it's an instantiated object, but static yeah